/*
 * Contains unit tests for testing [p5.string.split] in system
 */

/*
 * Tests that p5.string.split behaves as expected when given a simple string
 */
simple-string
  p5.string.split:thomas,hansen
    =:,
  if:x:/-/*?count
    !=:int:2
    or:x:/./-/0?name
      !=:thomas
    or:x:/./-/1?name
      !=:hansen
    throw:@"Assert error, [p5.string.split] misbehaved!"

/*
 * Tests that p5.string.split behaves as expected when given a single integer
 */
integer
  p5.string.split:thomashansen
    =:int:6
  if:x:/-/*?count
    !=:int:2
    or:x:/./-/0?name
      !=:thomas
    or:x:/./-/1?name
      !=:hansen
    throw:@"Assert error, [p5.string.split] misbehaved!"

/*
 * Tests that p5.string.split behaves as expected when given two integers
 */
two-integers
  p5.string.split:thomashansenjohn
    =:int:6
    =:int:12
  if:x:/-/*?count
    !=:int:3
    or:x:/./-/0?name
      !=:thomas
    or:x:/./-/1?name
      !=:hansen
    or:x:/./-/2?name
      !=:john
    throw:@"Assert error, [p5.string.split] misbehaved!"

/*
 * Tests that p5.string.split behaves as expected when given a simple string
 */
two-strings
  p5.string.split:thomas,hansen john
    =:,
    =:" "
  if:x:/-/*?count
    !=:int:3
    or:x:/./-/0?name
      !=:thomas
    or:x:/./-/1?name
      !=:hansen
    or:x:/./-/2?name
      !=:john
    throw:@"Assert error, [p5.string.split] misbehaved!"

/*
 * Tests that p5.string.split behaves as expected
 */
regex
  p5.string.split:thomas,hansen john,doe
    =:regex:/[, ]/
  if:x:/-/*?count
    !=:int:4
    or:x:/./-/0?name
      !=:thomas
    or:x:/./-/1?name
      !=:hansen
    or:x:/./-/2?name
      !=:john
    or:x:/./-/3?name
      !=:doe
    throw:@"Assert error, [p5.string.split] misbehaved!"

/*
 * Tests that p5.string.split behaves as expected
 */
two-regex-throw
  _result
  try
    p5.string.split:thomas
      =:regex:/[, ]/
      =:regex:/hom/
  catch
    set:x:/../*/_result?value
      src:success
  if:x:/../*/_result?value
    !=:success
    throw:@"Assert error, [p5.string.split] misbehaved and did not throw exception when expected!"
